### PR TITLE
627 refactor unit tests

### PR DESCRIPTION
**Pull request by nickbeaird**
Refactor the unit tests to match the model filename structure. For example the felony_class_a.py file now has a test_felony_class_a.py counterpart. 

Files were made based on the class type that was being tested. A small change was to assert that each test was returning a specific instance type. 

Tests that included multiple class types were left in their own test file.

### CODE REVIEW

File: src/backend/tests/models/charge_types/test_felony_class_b.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class TestSingleChargeConvictionsFelonyClassB(unittest.TestCase):
+    def setUp(self):
+        last_week = datetime.today() - timedelta(days=7)
+        self.single_charge = ChargeFactory.build(disposition=Disposition(ruling="Convicted", date=last_week))
+        self.charges = []
+
+    def create_recent_charge(self):
+        charge = ChargeFactory.save(self.single_charge)
```
Comment (loganballard) [inline src/backend/tests/models/charge_types/test_felony_class_b.py]: could this just be:

```
def create_recent_charge(self):
        return ChargeFactory.save(self.single_charge)
```

File: src/backend/tests/models/charge_types/test_felony_class_b.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        assert charge.expungement_result.type_eligibility.reason == "Further Analysis Needed"
+
+    def test_class_felony_is_added_to_b_felony_attribute(self):
+        self.single_charge["name"] = "Aggravated theft in the first degree"
+        self.single_charge["statute"] = "164.057"
+        self.single_charge["level"] = "Felony Class B"
+        charge = self.create_recent_charge()
+
+        assert isinstance(charge, FelonyClassB)
+        assert charge.__class__.__name__ == "FelonyClassB"
```
Comment (loganballard) [inline src/backend/tests/models/charge_types/test_felony_class_b.py]: This is probably my own ignorance, but is there a meaningful difference between the check on this line and the one above?

File: src/backend/tests/models/charge_types/test_felony_class_a.py [hunk 3]
```diff
@@ -2,13 +2,37 @@
...
+from expungeservice.models.charge_types.felony_class_a import FelonyClassA
+from expungeservice.models.disposition import Disposition
 from expungeservice.models.expungement_result import EligibilityStatus
 
 from tests.factories.charge_factory import ChargeFactory
-from expungeservice.models.disposition import Disposition
 
 
-class TestSingleChargeAcquittals(unittest.TestCase):
+class TestSingleChargeConvictionsFelonyClassA(unittest.TestCase):
```
Comment (loganballard) [inline src/backend/tests/models/charge_types/test_felony_class_a.py]: Seems like for Class A/B/C, all of these testing classes share the setup/create_recent_charge methods, should they be refactored into a generic `TestSingleChargeConvictionsFelony(unittest.TestCase)` base class, which then ``TestSingleChargeConvictionsFelonyClass{A/B/C}(`TestSingleChargeConvictionsFelony)` could inherent from?

File: src/backend/tests/models/charge_types/test_marijuana_ineligible.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+from datetime import datetime, timedelta
+
+from expungeservice.models.expungement_result import EligibilityStatus
+from expungeservice.models.charge_types.marijuana_ineligible import MarijuanaIneligible
+
+from tests.factories.charge_factory import ChargeFactory
+from expungeservice.models.disposition import Disposition
+
+
+class TestSingleChargeConvictionsMarijuanaIneligible(unittest.TestCase):
```
Comment (loganballard) [inline src/backend/tests/models/charge_types/test_marijuana_ineligible.py]: This one could inherent that "base testing case" class too

File: src/backend/tests/models/charge_types/test_marijuana_ineligible.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        return charge
+
+    def test_marijuana_ineligible_statute_475b3493c(self):
+        self.single_charge["name"] = "Unlawful Manufacture of Marijuana Item"
+        self.single_charge["statute"] = "475B.349(3)(C)"
+        self.single_charge["level"] = "Felony Class C"
+        marijuana_felony_class_c = self.create_recent_charge()
+        self.charges.append(marijuana_felony_class_c)
+
+        assert isinstance(marijuana_felony_class_c, MarijuanaIneligible)
```
Comment (loganballard) [inline src/backend/tests/models/charge_types/test_marijuana_ineligible.py]: Could all of these common asserts be a single helper method?

```
def check_marijuana_inelligibility():
        assert isinstance(marijuana_felony_class_a, MarijuanaIneligible)
        assert marijuana_felony_class_a.type_name == "Marijuana Ineligible"
        assert marijuana_felony_class_a.expungement_result.type_eligibility.status is EligibilityStatus.INELIGIBLE
        assert marijuana_felony_class_a.expungement_result.type_eligibility.reason == "Ineligible under 137.226"
```

File: src/backend/tests/models/charge_types/test_felony_class_b.py (hunk 2 above)
Comment (KentShikama) [reply src/backend/tests/models/charge_types/test_felony_class_b.py]: Just the latter is harder to read and navigate references with :P

**Comment by wittejm** [comment]
@nickbeaird I pulled your changes here and am working on them with my active work #844 , so I'm closing this PR.

**Pull request closed without merging**